### ISSUE TITLE
Alia somehow conflicts with clojurescript 1.9.456

### ISSUE THREAD

**Issue by piranha** [mpenet/alia#74]
If I create `project.clj` like that:

```
(defproject qwe "0.1.0-SNAPSHOT"
  :dependencies
  [[org.clojure/clojure "1.8.0"]
   [cc.qbits/alia-all "3.3.0"]
   [org.clojure/clojurescript "1.9.456"]])
```

then `(require 'qbits.alia)` fails with this in REPL:

```
CompilerException java.lang.NoClassDefFoundError: com/google/common/util/concurrent/FutureFallback, compiling:(qbits/alia/codec/udt.clj:110:12) 
```

Removing clojurescript from dependencies (or changing to an earlier version) removes this problem. I'm not entirely sure how to fix that, but I would love to see it fixed. :)

**Comment by mpenet** [comment]
I can have a look on monday, but it seems cljs ships with a more recent version of guava than java-driver, and the later overrides it. You can try to add an exclude for guava in your alia dependency in the project.clj file.

**Comment by mpenet** [comment]
Actually this is more tricky than I thought. 

cassandra-driver-core (the java driver from datastax) ships with guava 16.x, google closure compiler with 20.x. But as seen in https://datastax-oss.atlassian.net/browse/JAVA-1328 & https://github.com/datastax/java-driver/pull/769 both are not stricly compatible and we'll have to wait for them to release 3.1.4 it seems (they're talking about either shadowing guava or bumping it to 20.x. 

I think your best solution at the moment would be to use different profiles for cljs and clj development on your project and exclude alia from the cljs profile for instance

**Comment by piranha** [comment]
Right, thanks. I've tried playing with various versions of Guava and Closure Compiler, and had no luck. So it seems we'll be stuck with old CLJS for some time - it's a bit hard to separate in different profiles, since we're developing a single application with server rendering, so it's somewhat coupled.

I don't know if this should be closed or you want to left it open for others, I'm ok with either way. :)

**Comment by mpenet** [comment]
I did manage to get something running while playing with excludes when the
issue was created, but I didnt check if the cljs compilation was working. I
ll see if I can reproduce it tomorrow and post the setup here (it was
messy).

On Feb 6, 2017 22:19, "Alexander Solovyov" <notifications@github.com> wrote:

Right, thanks. I've tried playing with various versions of Guava and
Closure Compiler, and had no luck. So it seems we'll be stuck with old CLJS
for some time - it's a bit hard to separate in different profiles, since
we're developing a single application with server rendering, so it's
somewhat coupled.

I don't know if this should be closed or you want to left it open for
others, I'm ok with either way. :)

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/mpenet/alia/issues/74#issuecomment-277816651>, or mute
the thread
